courseraresearchexports/exports/utils.py

```diff
@@ -0,0 +1,223 @@
+    chunk_size = 1024 * 1024
+    output_filename = os.path.join(dest, export_filename)
+
+    with open(output_filename, 'wb') as f:
+        for data in tqdm(
+                iterable=response.iter_content(chunk_size),
+                total=int(response.headers['Content-length'])/chunk_size,
+                unit='MB',
+                desc=export_filename):
+
+            f.write(data)
+
+    return output_filename
+
+
+def lookup_course_slug_by_id(course_id):
+    """
+    Find the course slug given an course_id
+    """
+    response = requests.get(requests.compat.urljoin(COURSE_API, course_id))
+    response.raise_for_status()
+    return response.json()['elements'][0]['slug']
+
+
+def lookup_course_id_by_slug(course_slug):
+    """
+    Find the course_id given a course_slug
+    """
+    payload = {'q': 'slug', 'slug': course_slug}
+    response = requests.get(COURSE_API, params=payload)
+    response.raise_for_status()
+    return response.json()['elements'][0]['id']
+
+
+def lookup_partner_id_by_short_name(partner_short_name):
+    """
+    Find the partner_id by short name
+    """
+    payload = {'q': 'shortName', 'shortName': partner_short_name}
+    response = requests.get(COURSE_API, params=payload)
+    response.raise_for_status()
+    return response.json()['elements'][0]['id']
+
+
+def lookup_partner_short_name_by_id(partner_id):
+    """
+    Find the partner_id by short name
+    """
+    response = requests.get(requests.compat.urljoin(PARTNER_API, partner_id))
+    response.raise_for_status()
+    return response.json()['elements'][0]['slug']
+
+
+def get_scope_id_and_name_from_job(export_job):
+
+    job_scope = export_job['scope']
+    if job_scope['typeName'] == 'courseContext':
+        scope_id = job_scope['definition']['courseId']
+        scope_name = lookup_course_slug_by_id(scope_id)
+    elif job_scope['typeName'] == 'partnerContext':
+        scope_id = job_scope['definition']['partnerId']['maestroId']
+        scope_name = lookup_partner_short_name_by_id(scope_id)
+    elif job_scope['typeName'] == 'groupContext':
+        scope_id = job_scope['definition']['groupId']
+        scope_name = scope_id
+
+    return scope_id, scope_name
+
+
+def create_export_job_json(**kwargs):
+    """Creates json body for On Demand Research Exports request.
+    Keyword arguments:
+    courseID -- Unique identifier for course assigned by Coursera
+    courseSlug -- Unique name for course assigned by Coursera
+    partnerId -- Identifier for partner institution
+    partnerShortName -- Identifier for partner institution
+    groupId -- Identifier for group
+    exportType -- Currently only RESEARCH_WITH_SCHEMAS
+    schemaNames -- for RESEARCH_WITH_SCHEMAS datatypes to be exported
+    interval -- dates to export when exporting RESEARCH_EVENTING
+    anonymityLevel -- Level of user ID hashing
+    statementOfPurpose -- Statement for usage of this export data
+    """
+
+    request = {}
+    if kwargs.get('courseId'):
```

Comment:
Sure, will do.  The problem I encountered is that `ArgumentParser` will add `'courseId': None` even if it's not satisfied, so the key exists.  Would `kwargs['courseId']` be better?